services/gundeck/src/Gundeck/Push.hs

```diff
@@ -224,24 +223,7 @@ pushNative sendNotice notif p rcps
     | ntfTransient notif = if sendNotice
         then pushNotice p notif rcps
         else pushData p notif rcps
-    | otherwise = case partition (preferNotice sendNotice (p^.pushOrigin)) rcps of
-        (xs, []) -> pushNotice p notif xs
-        ([], ys) -> pushData p notif ys
-        (xs, ys) -> do
-            a <- async $ pushNotice p notif xs
-            pushData p notif ys
-            wait a
-
--- If a fallback address is set, the push is not transient, and
--- the address does not belong to the origin user, we prefer
--- type=notice notifications even for the first attempt, since the device
--- has to make a request to cancel the fallback notification in any
--- case, which it can combine with fetching the notification in a single
--- request. We can thus save the effort of encrypting and decrypting native
--- push payloads to such addresses.
-preferNotice :: Bool -> UserId -> Address s1 -> Bool
-preferNotice sendNotice orig a = sendNotice
-                                  || (isJust (a^.addrFallback) && orig /= (a^.addrUser))
```

Comment:
Did you really mean to remove the whole "`preferNotice`" block on this PR? If so, I am not sure I understand the reasoning :/

Reply:
I actually changed my mind about what I did here.  Now I *think* it is the same code as before, but I am going about this very syntactically, with very little understanding.

`addrFallback` should always be `False` now, so `preferNotice` can be simplified to `preferNotice bla _ _ = bla`.

This means that the `case partition...` in `pushNative` will always fall into one of the first two cases.  Rewriting that ends up with exactly the same code as in the `ntfTransient notif` case, so the guards can be eliminated completely.

Does this match your intuition of what the code is supposed to do?

Reply:
note that the `REFACTOR` comment is bogus because `sendNotice` will always be true after https://github.com/wireapp/wire-server/pull/518